Write tests for instruments.get/set methods

romandev opened the pull request:
This change covers basic behavior of get/set methods.

<!-- Reviewable:start -->

<!-- Reviewable:end -->

romandev committed:
Write tests for instruments.get/set methods

This change covers basic behavior of get/set methods.

w3c-bots commented:
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/21199)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/308908255)

romandev commented:
@marcoscaceres I've seen your comments in my another PR. (Thank you)
It's not related WebPayment, but I accidentally pushed it there. Sorry for the confusion.

Anyway, I addressed your comments in this change. We might always need manual tests due to requestPermission().

marcoscaceres commented on payment-handler/payment-instruments-manual.https.html:
```diff
@@ -0,0 +1,52 @@
+
+  // PaymentInstruments.get() should resolve `undefined` if no stored key
+  const undefinedInstrument = await instruments.get('undefined_key');
+  assert_equals(undefinedInstrument, undefined);
```
Probably want a test that overrides an existing entry. 

Also, we probably want a test to check what equality of pulling the same instrument multiple times out of storage.

Additionally, we probably want to check that these objects are just copies, in that if you change them, they don't change the one in storage. 

After changing one, you have to store it again.

marcoscaceres replied:
Oh, and we probably want to be mean... and add maybe 1024 instruments. And we want to check that the order is kept.

Also, what want to check that setting an existing key retains the order. So, setting key "test_instrument1" to a new instrument should still keep it as the first.

marcoscaceres reviewed (changes requested):
Great start on these... suggested a few more tests we should add.
